lib/npmdc/checkers/checker.rb

```diff
@@ -0,0 +1,52 @@
+require 'npmdc/formatter'
+require 'npmdc/checkers/npm/checker'
+require 'npmdc/checkers/yarn/checker'
+
+module Npmdc
+  module Checkers
+    class Checker
+      include Npmdc::Errors
+
+      attr_reader :path, :formatter, :types
+
+      def initialize(options)
+        @package_manager = options.fetch('package_manager', Npmdc.config.package_manager)
+        @path_to_yarn = options.fetch('path_to_yarn', Npmdc.config.path_to_yarn)
+        @path = options.fetch('path', Npmdc.config.path)
+        @types = options.fetch('types', Npmdc.config.types)
+        @abort_on_failure = options.fetch('abort_on_failure', Npmdc.config.abort_on_failure)
+        @formatter = Npmdc::Formatter.(options)
+      end
+
+      def call
+        checker =
+          if @package_manager == 'npm'
```

Comment:
I'm sure, we can resolve the checker adapter dynamically without this explicit condition, but it's the matter of taste.